istioctl/cmd/workload.go

```diff
@@ -0,0 +1,402 @@
+		Use:   "entry",
+		Short: "Commands dealing with WorkloadEntry resources",
+	}
+	entryCmd.AddCommand(configureCommand())
+	return entryCmd
+}
+
+func createCommand() *cobra.Command {
+	createCmd := &cobra.Command{
+		Use:   "create",
+		Short: "Creates a WorkloadGroup resource that provides a template for associated WorkloadEntries",
+		Long: `Creates a WorkloadGroup resource that provides a template for associated WorkloadEntries.
+The default output is serialized YAML, which can be piped into 'kubectl apply -f -' to send the artifact to the API Server.`,
+		Example: "create --name foo --namespace bar --labels app=foo,bar=baz --ports grpc=3550,http=8080 --serviceAccount sa",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if name == "" {
+				return fmt.Errorf("expecting a workload name")
+			}
+			if namespace == "" {
+				return fmt.Errorf("expecting a workload namespace")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			u := &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": collections.IstioNetworkingV1Alpha3Workloadgroups.Resource().APIVersion(),
+					"kind":       collections.IstioNetworkingV1Alpha3Workloadgroups.Resource().Kind(),
+					"metadata": map[string]interface{}{
+						"name":      name,
+						"namespace": namespace,
+					},
+				},
+			}
+			spec := &networkingv1alpha3.WorkloadGroup{
+				Metadata: &networkingv1alpha3.WorkloadGroup_ObjectMeta{
+					Labels: convertToStringMap(labels),
+				},
+				Template: &networkingv1alpha3.WorkloadEntry{
+					Ports:          convertToUnsignedInt32Map(ports),
+					ServiceAccount: serviceAccount,
+				},
+			}
+			wgYAML, err := generateWorkloadGroupYAML(u, spec)
+			if err != nil {
+				return err
+			}
+			_, err = cmd.OutOrStdout().Write(wgYAML)
+			return err
+		},
+	}
+	createCmd.PersistentFlags().StringVar(&name, "name", "", "The name of the workload group")
+	createCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", "", "The namespace that the workload instances will belong to")
+	createCmd.PersistentFlags().StringSliceVarP(&labels, "labels", "l", nil, "The labels to apply to the workload instances; e.g. -l env=prod,vers=2")
+	createCmd.PersistentFlags().StringSliceVarP(&ports, "ports", "p", nil, "The incoming ports exposed by the workload instance")
+	createCmd.PersistentFlags().StringVarP(&serviceAccount, "serviceAccount", "s", "default", "The service identity to associate with the workload instances")
+	return createCmd
+}
+
+func generateWorkloadGroupYAML(u *unstructured.Unstructured, spec *networkingv1alpha3.WorkloadGroup) ([]byte, error) {
+	iSpec, err := unstructureIstioType(spec)
+	if err != nil {
+		return nil, err
+	}
+	u.Object["spec"] = iSpec
+
+	wgYAML, err := yaml.Marshal(u.Object)
+	if err != nil {
+		return nil, err
+	}
+	return wgYAML, nil
+}
+
+func configureCommand() *cobra.Command {
+	configureCmd := &cobra.Command{
+		Use:   "configure",
```

Comment:
The command name `workload entry configure` seems a bit off to me - but I am not an expert and can offer no better ideas. @esnible is and likely can though, as I recall a lot of discussion about the grammar/structure of command names